### ISSUE TITLE
Updating from 0.8.0 to 0.9.0 causes SwiftUI animations to lag.

### ISSUE THREAD

**Issue by megabitsenmzq** [groue/GRDBQuery#54]
| Before    | After |
| -------- | ------- |
| <video style="width:100%" src="https://github.com/user-attachments/assets/c2f11979-1748-465e-933d-b669245c7c1e" type="video/mp4">|<video style="width:100%" src="https://github.com/user-attachments/assets/a4a17e0d-a78f-47b3-b470-b0a087feb20e" type="video/mp4"> |

What I did:

1. Update from 0.8.0 to 0.9.0.
2. `Queryable` to `ValueObservationQueryable`
3. `func fetchValue(_ db: Database) throws` to `func fetch(_ db: Database) throws`
4. `.environment(\.appDatabase, AppDatabase.shared)` to `.appDatabase(.shared)`

Basically, things I need to do according to the migration guide.

**Comment by megabitsenmzq** [comment]
The animation is `matchedGeometryEffect`, and it won't cause any data changes.

**Comment by groue** [comment]
Hello @megabitsenmzq,

Can you please share your previous (`Queryable`) and your new code (`ValueObservationQueryable`)?

**Comment by megabitsenmzq** [comment]
I just built the app with Xcode 15 and the problem seems gone. It might be an Xcode Beta problem. But it only happened after I updated GRDBQuery.

```
struct SlitPhotoItemRequest: ValueObservationQueryable {
    enum Ordering {
        case byDate
    }
    
    var ordering: Ordering

    static var defaultValue: [SlitPhotoItem] { [] }
    
    func fetch(_ db: Database) throws -> [SlitPhotoItem] {
        switch ordering {
        case .byDate:
            return try SlitPhotoItem.all().orderedByDate().fetchAll(db)
        }
    }
}
```

```
struct SlitPhotoItemRequest: Queryable {
    enum Ordering {
        case byDate
    }
    
    var ordering: Ordering
    
    static var defaultValue: [SlitPhotoItem] { [] }
    
    func publisher(in appDatabase: AppDatabase) -> AnyPublisher<[SlitPhotoItem], Error> {
        ValueObservation
            .tracking(fetchValue(_:))
            .publisher(
                in: appDatabase.reader,
                scheduling: .immediate)
            .eraseToAnyPublisher()
    }
    
    func fetchValue(_ db: Database) throws -> [SlitPhotoItem] {
        switch ordering {
        case .byDate:
            return try SlitPhotoItem.all().orderedByDate().fetchAll(db)
        }
    }
}
```

**Comment by groue** [comment]
Thank you, @megabitsenmzq. Both codes should be equivalent. 🤔 

May I ask for more of your help? If you revert `SlitPhotoItemRequest` back to Queryable, as below, what happens? (Take care of `publisher(in:)` argument, now of type `DatabaseContext`):

```swift
struct SlitPhotoItemRequest: Queryable {
    enum Ordering {
        case byDate
    }
    
    var ordering: Ordering
    
    static var defaultValue: [SlitPhotoItem] { [] }
    
    func publisher(in context: DatabaseContext) -> AnyPublisher<[SlitPhotoItem], Error> {
        ValueObservation
            .tracking(fetchValue(_:))
            .publisher(
                in: context,
                scheduling: .immediate)
            .eraseToAnyPublisher()
    }
    
    func fetchValue(_ db: Database) throws -> [SlitPhotoItem] {
        switch ordering {
        case .byDate:
            return try SlitPhotoItem.all().orderedByDate().fetchAll(db)
        }
    }
}
```

**Comment by megabitsenmzq** [comment]
I got an error "Type 'any DatabaseReader' cannot conform to 'DatabaseReader'". What should I do? 🤔

**Comment by groue** [comment]
Oh, I'm sorry. Please hold on until I come back with a better strategy.

Meanwhile, maybe check if other changes in the app could be the cause of the new behavior.

**Comment by megabitsenmzq** [comment]
> Oh, I'm sorry. Please hold on until I come back with a better strategy.
> 
> Meanwhile, maybe check if other changes in the app could be the cause of the new behavior.

Got it! Thank you for replying to me that fast. The only change in this commit is GRDBQuery. Here is a patch file you can check.

[LinearCCD-f67980f-Migrate to GRDBQuery 0.9..patch](https://github.com/user-attachments/files/16394215/LinearCCD-f67980f-Migrate.to.GRDBQuery.0.9.patch)

**Comment by megabitsenmzq** [comment]
I eventually found this was my own fault. I put some large image files within the body, which caused it to load every time the view refreshed.

**Comment by groue** [comment]
Oh, thank you @megabitsenmzq 🙏. I have been slow to answer, but I did not forget this issue. I'm happy the problem was solved application-side!
